src/ray/core_worker/core_worker.cc

```diff
@@ -1799,7 +1801,18 @@ void CoreWorker::HandleWaitForActorOutOfScope(
   if (it == actor_handles_.end()) {
     respond(actor_id);
   } else {
-    RAY_CHECK(actor_out_of_scope_callbacks_.emplace(actor_id, std::move(respond)).second);
+    auto object_id_refs = reference_counter_->GetObjectIdRefs();
+    const auto actor_creation_return_id = ObjectID::ForActorHandle(actor_id);
+    auto iter = object_id_refs.find(actor_creation_return_id);
+    if (iter != object_id_refs.end() && iter->second.wait_for_delete) {
+      RAY_LOG(INFO)
```

Comment:
There is a race condition here because the ref can get deleted after we check if the ID is in the map.

Also, this is just generally a bad idea because accessing the hashmap is not threadsafe.